probe-rs/src/probe/jlink/mod.rs

```diff
@@ -898,28 +914,56 @@ impl DAPAccess for JLink {
                 // If ack[1] is set the host must retry the request. So let's do that right away!
                 log::debug!("DAP WAIT, retries remaining {}.", 5 - retry);
 
+                // Because we use overrun detection, we now have to clear the overrun error
+                let mut abort = Abort(0);
+
+                abort.set_orunerrclr(true);
+
+                DAPAccess::write_register(
+                    self,
+                    PortType::DebugPort,
+                    Abort::ADDRESS as u16,
+                    abort.into(),
+                )?;
+
                 continue;
             }
             if ack[2] {
+                log::debug!("DAP FAULT");
+
                 // A fault happened during operation.
 
                 // To get a clue about the actual fault we read the ctrl register,
                 // which will have the fault status flags set.
-
-                // Reading ctrl directly fails with nack on jlink. A dummy read
-                // of id first seems to clear it up.
-                let dp =
```

Comment:
do you know what specifically fixed this line? of it its just overrun support, what tipped you off to use overruns to fix it?

Reply:
I looked at a trace of my logic analyzer, and there the SWD analyzer seemed to be out of sync, as soon as the first WAIT response appeared. So no transfers after the WAIT response seemed correct.

## Avoiding sync problems
Based on the ARM spec, I assumed that this was because we always have a data phase after receiving the acknowledge bits, even if an error occurs. According to the spec, there should be no data phase if the response is WAIT or FAULT. But the JLink doesn't really allow us to do this, because we have to specify the whole transfer in advance. 

If we enable the overrun support, then it isn't a problem anymore that there is a data phase in case of an error, and probe and target don't get out of sync.

## Recovery when out of sync
Howevery, when we send an unexpected data phase, then the target enters a *protocol error state*. As long as it stays in that mode, it won't send any response anymore. To get out from this, you can either read the DPIDR register (what you did), or perform a line reset. The documentation says that the line reset is preferred, so I changed the code to only use a line reset in this case.

Reply:
ah its the reset. Beautiful work, thanks.